libs/blocks/chart/chart.js

```diff
@@ -0,0 +1,469 @@
+  const previousIsLarge = previousSize === LARGE;
+  const currentIsLarge = currentSize === LARGE;
+  const chartWrapper = el?.querySelector('.chart_wrapper');
+  const chartInstance = window.echarts?.getInstanceByDom(chartWrapper);
+
+  if (currentSize !== previousSize) {
+    updateContainerSize(chartWrapper, currentSize, chartType);
+    el.setAttribute('data-responsive-size', currentSize);
+  }
+
+  if (previousIsLarge !== currentIsLarge) {
+    chartInstance?.dispose();
+    initChart(chartWrapper, chartType, data, colors, currentSize);
+  } else {
+    chartInstance?.resize();
+  }
+};
+
+const init = async (el) => {
+  const children = el?.querySelectorAll(':scope > div');
+  const chartWrapper = children[2]?.querySelector(':scope > div');
+  children[0]?.classList.add('title');
+  children[1]?.classList.add('subtitle');
+  children[3]?.classList.add('footnote');
+  chartWrapper?.classList.add('chart_wrapper');
+
+  const chartStyles = el?.classList;
+  const section = el?.parentElement?.matches('.section') ? el.parentElement : null;
+  const sectionChildren = section?.querySelectorAll(':scope > div:not(.section-metadata)');
+  const upNumber = sectionChildren?.length;
+  section?.classList.add(`up-${upNumber}`);
+  section?.classList.add(SECTION_CLASSNAME);
+
+  let authoredSize = SMALL;
+  if (upNumber === 1) authoredSize = LARGE;
+  if (upNumber === 2) authoredSize = MEDIUM;
+
+  const size = getResponsiveSize(authoredSize);
+  el.classList.add(authoredSize);
+  el.setAttribute('data-responsive-size', size);
+
+  const chartType = chartTypes?.find((type) => el?.className?.indexOf(type) !== -1);
+  const dataLink = chartWrapper?.querySelector('a[href$="json"]');
+
+  dataLink?.remove();
+
+  if (!chartType || !chartWrapper || !dataLink) return;
+
+  updateContainerSize(chartWrapper, size, chartType);
+
+  if (chartType !== 'oversizedNumber') {
+    Promise.all([fetchData(dataLink), loadScript('/libs/deps/echarts.common.min.js')])
+      .then((values) => {
+        const data = values[0];
+
+        if (!data) return;
+
+        const authoredColor = Array.from(chartStyles)?.find((style) => style in colorPalette);
+        const hasOverride = Object.keys(data?.data[0])?.some((header) => header.toLowerCase() === 'color');
+        const colors = hasOverride
+          ? getOverrideColors(authoredColor, data.data)
+          : getColors(authoredColor);
+
+        if (!(window.IntersectionObserver)) {
+          initChart(chartWrapper, chartType, data, colors, size);
+        } else {
+          const observerOptions = {
+            root: null,
+            rootMargin: '0px',
+            threshold: 0.5,
+          };
+
+          const observer = new IntersectionObserver(
+            handleIntersect(chartWrapper, chartType, data, colors, size),
+            observerOptions,
+          );
+          observer.observe(el);
+        }
+
+        window.addEventListener('resize', throttle(
```

Comment:
It's worth looking into using the match media API which is way more performant than resize and innerwidth. You don't have to throttle as a result, too.

Top Google search result: https://webdevetc.com/blog/matchmedia-events-for-window-resizes/

Happy to help answer any questions if you run into them.